File: backend/api/masters.php
<?php
/**
 * API для получения данных о мастерах
 */

require_once 'config.php';
header('Content-Type: application/json');
header('Access-Control-Allow-Origin: *');
header('Access-Control-Allow-Methods: GET, POST, PUT, DELETE, OPTIONS');
header('Access-Control-Allow-Headers: Content-Type');

if ($_SERVER['REQUEST_METHOD'] === 'OPTIONS') {
    exit(0);
}

$pdo = getDBConnection();

// Получение всех мастеров
if ($_SERVER['REQUEST_METHOD'] === 'GET' && !isset($_GET['id'])) {
    try {
        $stmt = $pdo->query("SELECT * FROM masters ORDER BY created_at DESC");
        $masters = $stmt->fetchAll();
        
        // Для каждого мастера получаем его работы
        foreach ($masters as &$master) {
            $stmt = $pdo->prepare("SELECT * FROM master_works WHERE master_id = ? ORDER BY created_at DESC");
            $stmt->execute([$master['id']]);
            $master['works'] = $stmt->fetchAll();
        }
        
        echo json_encode(['success' => true, 'data' => $masters]);
    } catch (PDOException $e) {
        http_response_code(500);
        echo json_encode(['success' => false, 'message' => 'Ошибка сервера']);
    }
    exit;
}

// Получение одного мастера
if ($_SERVER['REQUEST_METHOD'] === 'GET' && isset($_GET['id'])) {
    try {
        $stmt = $pdo->prepare("SELECT * FROM masters WHERE id = ?");
        $stmt->execute([$_GET['id']]);
        $master = $stmt->fetch();
        
        if (!$master) {
            http_response_code(404);
            echo json_encode(['success' => false, 'message' => 'Мастер не найден']);
            exit;
        }
        
        $stmt = $pdo->prepare("SELECT * FROM master_works WHERE master_id = ? ORDER BY created_at DESC");
        $stmt->execute([$master['id']]);
        $master['works'] = $stmt->fetchAll();
        
        echo json_encode(['success' => true, 'data' => $master]);
    } catch (PDOException $e) {
        http_response_code(500);
        echo json_encode(['success' => false, 'message' => 'Ошибка сервера']);
    }
    exit;
}

// Создание мастера (требуется авторизация)
if ($_SERVER['REQUEST_METHOD'] === 'POST') {
    // Здесь должна быть проверка авторизации
    try {
        $data = json_decode(file_get_contents('php://input'), true);
        
        if (empty($data['name'])) {
            http_response_code(400);
            echo json_encode(['success' => false, 'message' => 'Имя обязательно']);
            exit;
        }
        
        $stmt = $pdo->prepare("INSERT INTO masters (name, description, photo, instagram, telegram, vk) VALUES (?, ?, ?, ?, ?, ?)");
        $stmt->execute([
            $data['name'],
            $data['description'] ?? null,
            $data['photo'] ?? null,
            $data['instagram'] ?? null,
            $data['telegram'] ?? null,
            $data['vk'] ?? null
        ]);
        
        echo json_encode(['success' => true, 'id' => $pdo->lastInsertId()]);
    } catch (PDOException $e) {
        http_response_code(500);
        echo json_encode(['success' => false, 'message' => 'Ошибка сервера']);
    }
    exit;
}

// Обновление мастера
if ($_SERVER['REQUEST_METHOD'] === 'PUT' && isset($_GET['id'])) {
    try {
        $data = json_decode(file_get_contents('php://input'), true);
        
        $stmt = $pdo->prepare("UPDATE masters SET name = ?, description = ?, photo = ?, instagram = ? WHERE id = ?");
        $stmt->execute([
            $data['name'],
            $data['description'] ?? null,
            $data['photo'] ?? null,
            $data['instagram'] ?? null,
            $_GET['id']
        ]);
        
        echo json_encode(['success' => true]);
    } catch (PDOException $e) {
        http_response_code(500);
        echo json_encode(['success' => false, 'message' => 'Ошибка сервера']);
    }
    exit;
}

// Удаление мастера
if ($_SERVER['REQUEST_METHOD'] === 'DELETE' && isset($_GET['id'])) {
    try {
        $stmt = $pdo->prepare("DELETE FROM masters WHERE id = ?");
        $stmt->execute([$_GET['id']]);
        
        echo json_encode(['success' => true]);
    } catch (PDOException $e) {
        http_response_code(500);
        echo json_encode(['success' => false, 'message' => 'Ошибка сервера']);
    }
    exit;
}

http_response_code(405);
echo json_encode(['success' => false, 'message' => 'Метод не разрешен']);
?>

File: backend/routes/masters.py
"""
API маршруты для управления мастерами
"""
from flask import Blueprint, request, jsonify
from models import db, Master, MasterWork

masters_bp = Blueprint('masters', __name__)


@masters_bp.route('/masters', methods=['GET', 'OPTIONS'])
@masters_bp.route('/masters/<int:master_id>', methods=['GET', 'OPTIONS'])
def get_masters(master_id=None):
    """Получение всех мастеров или одного мастера по ID"""
    if request.method == 'OPTIONS':
        return '', 200
    
    try:
        if master_id:
            # Получение одного мастера
            master = Master.query.get(master_id)
            if not master:
                return jsonify({'success': False, 'message': 'Мастер не найден'}), 404
            
            works = MasterWork.query.filter_by(master_id=master_id).order_by(MasterWork.created_at.desc()).all()
            data = master.to_dict(include_works=True)
            return jsonify({'success': True, 'data': data})
        else:
            # Получение всех мастеров
            masters = Master.query.order_by(Master.created_at.desc()).all()
            result = []
            for master in masters:
                master_data = master.to_dict(include_works=True)
                result.append(master_data)
            
            return jsonify({'success': True, 'data': result})
    
    except Exception as e:
        return jsonify({'success': False, 'message': 'Ошибка сервера'}), 500


@masters_bp.route('/masters', methods=['POST', 'OPTIONS'])
def create_master():
    """Создание нового мастера"""
    if request.method == 'OPTIONS':
        return '', 200
    
    try:
        data = request.get_json()
        
        if not data or not data.get('name'):
            return jsonify({'success': False, 'message': 'Имя обязательно'}), 400
        
        master = Master(
            name=data['name'],
            description=data.get('description'),
            photo=data.get('photo'),
            instagram=data.get('instagram'),
            telegram=data.get('telegram'),
            vk=data.get('vk')
        )
        
        db.session.add(master)
        db.session.commit()
        
        return jsonify({'success': True, 'id': master.id}), 201
    
    except Exception as e:
        db.session.rollback()
        return jsonify({'success': False, 'message': 'Ошибка сервера'}), 500


@masters_bp.route('/masters/<int:master_id>', methods=['PUT', 'OPTIONS'])
def update_master(master_id):
    """Обновление мастера"""
    if request.method == 'OPTIONS':
        return '', 200
    
    try:
        master = Master.query.get(master_id)
        if not master:
            return jsonify({'success': False, 'message': 'Мастер не найден'}), 404
        
        data = request.get_json()
        
        master.name = data.get('name', master.name)
        master.description = data.get('description', master.description)
        master.photo = data.get('photo', master.photo)
        master.instagram = data.get('instagram', master.instagram)
        master.telegram = data.get('telegram', master.telegram)
        master.vk = data.get('vk', master.vk)
        
        db.session.commit()
        
        return jsonify({'success': True})
    
    except Exception as e:
        db.session.rollback()
        return jsonify({'success': False, 'message': 'Ошибка сервера'}), 500


@masters_bp.route('/masters/<int:master_id>', methods=['DELETE', 'OPTIONS'])
def delete_master(master_id):
    """Удаление мастера"""
    if request.method == 'OPTIONS':
        return '', 200
    
    try:
        master = Master.query.get(master_id)
        if not master:
            return jsonify({'success': False, 'message': 'Мастер не найден'}), 404
        
        db.session.delete(master)
        db.session.commit()
        
        return jsonify({'success': True})
    
    except Exception as e:
        db.session.rollback()
        return jsonify({'success': False, 'message': 'Ошибка сервера'}), 500
